xbmc/windowing/wayland/XkbcommonKeymap.cpp

```diff
@@ -221,10 +283,21 @@ std::unique_ptr<CXkbcommonKeymap> CXkbcommonContext::KeymapFromString(std::strin
     throw std::runtime_error("Failed to compile keymap");
   }
 
-  return std::make_unique<CXkbcommonKeymap>(std::move(xkbKeymap));
+  std::unique_ptr<xkb_compose_table, CXkbcommonKeymap::XkbComposeTableDeleter> xbkComposeTable{
+      xkb_compose_table_new_from_locale(m_context.get(), locale.c_str(),
+                                        XKB_COMPOSE_COMPILE_NO_FLAGS),
+      CXkbcommonKeymap::XkbComposeTableDeleter()};
+
+  if (!xbkComposeTable)
+  {
+    throw std::runtime_error("Failed to compile localized compose table");
```

Comment:
I am hitting this on webOS probably because `/usr/share/X11/locale`does not exist breaking keyboard input completely due to missing keymap. Can we not error out here early and return `nullptr` for the compose map instead?

Reply:
Yeah that's why I asked for the review,  I'll try to make this optional :)
Added logger configuration on a new commit, I guess you're right in your assumption:

```
2023-10-16 21:40:23.658 T:21155   debug <general>: xkbLogger: Compiling xkb_keycodes "(unnamed)"
2023-10-16 21:40:23.658 T:21155   debug <general>: xkbLogger: Compiling xkb_types "(unnamed)"
2023-10-16 21:40:23.658 T:21155   debug <general>: xkbLogger: Compiling xkb_compatibility "(unnamed)"
2023-10-16 21:40:23.658 T:21155   debug <general>: xkbLogger: Compiling xkb_symbols "(unnamed)"
2023-10-16 21:40:23.660 T:21155   debug <general>: xkbLogger: created compose table from locale C with path /usr/share/X11/locale/en_US.UTF-8/Compose
```

Reply:
@sundermann https://github.com/xbmc/xbmc/pull/23943/commits/59f8bdce0768beb64480454d653bd91931890197 makes this optional so it should no longer crash on webOS.
Still it'd be nice if you could look it up. I guess the compose files must exist somewhere in webOS (and we also can use libxkbcommon to load the table from a given file).

Reply:
Can confirm keyboard / remote support is working again. I tried mounting `/usr/share/X11/locale` from `libX11-data` from debian but for some reason it would still not read from these files. I also tried reading directly from the compose file using `xkb_compose_table_new_from_file` which worked but dead keys were still not working. I wouldn't worry too much about it though, probably not implemented properly in LG's wayland compositor